curvefs/src/metaserver/storage/rocksdb_storage.h

```diff
@@ -335,13 +339,21 @@ class RocksDBStorageIterator : public Iterator {
           ordered_(ordered) {
         if (status_ == 0) {
             readOptions_ = storage_->ReadOptions();
-            readOptions_.snapshot = storage_->db_->GetSnapshot();
+            if (storage_->InTransaction_) {
+                readOptions_.snapshot = storage_->txn_->GetSnapshot();
+            } else {
+                readOptions_.snapshot = storage_->db_->GetSnapshot();
+            }
```

Comment:
> what's the difference between in transaction and not in transaction?

if we use `db_->GetSnapshot()` to get snapshot , we will can't access the keys which set by transaction, consider the following situation:
```
txn = kvStorage->BeginTransaction();
txn->Set("key1", "value1");
Status s = txn->Get("key1");  // it will return Status:NotFound()
txn->Commit();
```